src/org/labkey/test/components/ui/entities/EntityInsertPanel.java

```diff
@@ -196,6 +264,64 @@ protected boolean isVisible(Locator locator)
         }
     }
 
+    public EntityInsertPanel showGrid()
+    {
+        if (!isGridVisible())
+            elementCache().modeSelectListItem("from Grid")
+                    .waitForElement(this, 2000).click();
+        clearElementCache();
+        WebDriverWrapper.waitFor(()-> isGridVisible(),
+                "the grid did bot become visible", 2000);
```

Comment:
Hide this all under the `if` so that we don't clear the cache unnecessarily.
```suggestion
        if (!isGridVisible())
        {
            elementCache().modeSelectListItem("from Grid")
                    .waitForElement(this, 2000).click();
            clearElementCache();
            WebDriverWrapper.waitFor(()-> isGridVisible(),
                "the grid did bot become visible", 2000);
        }
```